src/main/java/frc/robot/trajectory/Kinematics.java

```diff
@@ -0,0 +1,126 @@
+package frc.robot.trajectory;
+
+import frc.robot.Constants;
+
+public class Kinematics {
+
+	public static Point updateLineOdometry(double gyroAngle, double currentEncoderPos, double prevEncoderPos, Point robotPos) {
+		// double currentEncoderPos = ((-leftEncoderPos + rightEncoderPos) / 2.0);
+		double dEncoder = (currentEncoderPos - prevEncoderPos) / Constants.REVOLUTIONS_PER_INCH;
+		double dX = dEncoder * Math.cos(Math.toRadians(gyroAngle));
+		double dY = dEncoder * Math.sin(Math.toRadians(gyroAngle));
+
+		return new Point(robotPos.getX() + dX, robotPos.getY() + dY);
+	}
+
+	public static Point updateArcOdometry(double gyroAngle, double prevGyroAngle, double currentEncoderPos, double prevEncoderPos, Point robotPos) {
+		double theta = Math.abs(gyroAngle - prevGyroAngle);
+		double arcLength = (currentEncoderPos - prevEncoderPos) / Constants.REVOLUTIONS_PER_INCH;
+		if (Math.abs(theta) < Constants.ODOMETRY_MIN_THETA) {
+			theta = Constants.ODOMETRY_MIN_THETA;
+		}
+		double radius = 180 * arcLength / (Math.PI * theta);
+		double alpha = prevGyroAngle - 90;
+		double circleX = robotPos.getX() + radius * Math.cos(Math.toRadians(alpha));
+		double circleY = robotPos.getY() + radius * Math.sin(Math.toRadians(alpha));
+		double beta = alpha + 180 - theta;
+		double robotNewXPos = circleX + radius * Math.cos(Math.toRadians(beta));
+		double robotNewYPos = circleY + radius * Math.sin(Math.toRadians(beta));
+		
+		return new Point(robotNewXPos, robotNewYPos);
+	}
+
+	public static Point inversekinematics(double gyroHeading, Point robotPos, Point targetPos) {
+		Point newtargetPos = new Point(targetPos.getX() - robotPos.getX(), targetPos.getY() - robotPos.getY());
+
+		double mRC = Math.tan(Math.toRadians(gyroHeading));
```

Comment:
Can you rename these single letter variables, or add comments on what these represent? Also the units.